### PR TITLE
feat(cli): initialize apps based on docker builds

### PR DESCRIPTION
**Pull request by jfeodor**

### CODE REVIEW

File: cmd/init/cmd.go [hunk 1]
```diff
@@ -22,27 +24,44 @@ var InitCmd = &cobra.Command{
...
+		AppDir:           appDir,
+		Name:             argName,
+		Desc:             argDesc,
+		LibraryKey:       argLibraryKey,
+		AppFile:          argAppFile,
+		RequirementsFile: argRequirementsFile,
+		Dockerfile:       argDockerfile,
+		DockerContext:    argDockerContext,
+	}
+	_, err = Initialize(&wizard.SurveyAsker{}, params)
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply cmd/init/cmd.go:62]: This return value is used in a different place.

File: internal/manifest/file.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const ManifestFileName string = "numerous.toml"
+
+var ManifestPath string = filepath.Join(".", ManifestFileName)
+
+func Exists(appDir string) (bool, error) {
+	manifestPath := filepath.Join(appDir, ManifestFileName)
+	_, err := os.Stat(manifestPath)
+	exists := !errors.Is(err, os.ErrNotExist)
+
+	return exists, err
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply internal/manifest/file.go]: This method was not used anywhere and is removed in 4101e9cd2da2577181db9dedd74a871aca2e2ebb

File: cmd/init/initialize.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	if m.Python != nil && m.Python.Library.Key != "" {
+		m.Python.Version = python.PythonVersion()
+	}
+
+	err = m.BootstrapFiles("", params.AppDir)
+	switch {
+	case err == manifest.ErrEncodingManifest:
+		output.PrintErrorDetails("Error encoding manifest file", err)
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply cmd/init/initialize.go:79]: Fixed in ae7e67cb06e3951d78c57a620e0bd155c4b39fe3

File: internal/manifest/files.go [hunk 4]
```diff
@@ -11,17 +11,30 @@ import (
 	"numerous.com/cli/internal/dir"
 )
 
+func fileExists(path string) (bool, error) {
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply internal/manifest/files.go:14]: The other function was removed

File: internal/manifest/bootstrap_docker.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+	if err := createAndWriteIfFileNotExist(dockerfilePath, dockerExampleDockerfile); err != nil {
+		return err
+	}
+
+	if err := createAndWriteIfFileNotExist(appPath, dockerExampleAppPy); err != nil {
+		return err
+	}
+
+	if err := createAndWriteIfFileNotExist(requirementsPath, dockerExampleRequirementsTxt); err != nil {
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply internal/manifest/bootstrap_docker.go:60]: The idea is that we bootstrap an example setup with a Dockerfile if a file does not already exist.

If a Dockerfile already exists, the function just returns `ErrNoBootstrapDockerfileExists`

File: internal/manifest/bootstrap.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	if toolID != "" {
+		gitignoreLines = append(gitignoreLines, dir.AppIDFileName)
+	}
+
+	if err := addToGitIgnore(basePath, gitignoreLines); err != nil {
+		return err
+	}
+
+	switch {
+	case m.Python != nil && m.Python.Library.Key != "":
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply internal/manifest/bootstrap.go]: 3a671f8

File: cmd/init/initialize.go [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+			Context:    params.DockerContext,
+		},
+	}
+	m, err := wizard.Run(asker, runWizardParams)
+	if err != nil {
+		output.PrintErrorDetails("Error running initialization wizard", err)
+		return nil, err
+	}
+
+	if m.Python != nil && m.Python.Library.Key != "" {
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply cmd/init/initialize.go:72]: yes, 3a671f863230b34198a8c9da85477d7987191e4e fixes it. It works as expected.

File: cmd/init/initialize.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+			"The specified library %s is not supported. Supported libraries are %s.",
+			params.LibraryKey,
+			manifest.SupportedLibraryValuesList(),
+		)
+
+		return nil, err
+	}
+
+	runWizardParams := wizard.RunWizardParams{
+		ProjectFolderPath: params.AppDir,
```
[the remark this answers is not in the document]
Comment (jfeodor) [reply cmd/init/initialize.go:49]: The wizard should not care about projects, so the variable is just misnamed.

[the remark this answers is not in the document]
Comment (jfeodor) [reply cmd/init/initialize.go]: Fixed in ff9732c

File: internal/manifest/bootstrap_docker.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (moroderNumerous) [reply internal/manifest/bootstrap_docker.go:59]: AAaah I see, that check happens way before this!

**Review by moroderNumerous — APPROVED** [review]

**Pull request merged**